File: Go/Algorithms/2154.go
package main

func findFinalValue(nums []int, original int) int {
	s := make(map[int]bool)
	for _, el := range nums {
		s[el] = true
	}
	// ugh
	for _, ok := s[original]; ok; _, ok = s[original] {
		original *= 2
	}
	return original
}

File: Rust/Algorithms/2154.rs
use std::collections::HashSet;
use std::iter::FromIterator;

impl Solution {
    pub fn find_final_value(nums: Vec<i32>, mut original: i32) -> i32 {
        let s: HashSet<i32> = nums.into_iter().collect();
        while s.contains(&original) {
            original *= 2;            
        }
        original
    }
}